### ISSUE TITLE
Ensure WooCommerce is installed after purchasing the eCommerce plan

### ISSUE THREAD

**Issue by daledupreez** [Automattic/wp-calypso#75951]
### Quick summary

After purchasing the eCommerce plan via many of our flows, including from `wordpress.com/start/`, we show some processing screens while we wait for the site to transfer to the Atomic platform. However, we don't wait for all eCommerce plugins to be installed, so we can hit timing gaps where site's plugins are not fully set up yet.

This was originally reported internally here: p1681496180387279-slack-C029GN3KD, which led us to create #75796, which we addressed yesterday via #75835. However, that fix only handles the case where we're sending users through our newer eCommerce-specific flow, and not the catch-all case where users have just purchased the eCommerce plan via some other flow(s).

### Steps to reproduce

1. While logged in, visit (https://wordpress.com/start/)
2. On the domains step, click on the option to choose a domain later
3. On the plans step, pick the Commerce/eCommerce plan
4. Complete checkout to purchase the plan
5. Wait for the loading screens to complete, and get ready to click the CTA to "Create your store" as soon as it appears
6. When the "Thank you for your purchase" screen appears, click on the CTA as quickly as you can
7. Sometimes (but not always), this triggers an error, because the site is not fully set up yet, either because Jetpack SSO isn't configured yet, or because WooCommerce is not installed and activated yet

### What you expected to happen

No errors should appear and I should be redirected to the WooCommerce home page.

### What actually happened

Sometimes, users may see one of the following errors:
* "Error: Missing or invalid site id" - this is an error from Jetpack indicating SSO is not correctly configured yet
* "Sorry, you are not allowed to access this page". - this is an error that can be displayed when the WooCommerce plugin is not yet activated or fully set up

### Impact

Some (< 50%)

### Available workarounds?

Yes, easy to implement

### Platform (Simple and/or Atomic)

Atomic

**Comment by daledupreez** [comment]
I believe that we need to add some logic to check for installed plugins somewhere near this code: https://github.com/Automattic/wp-calypso/blob/ae08b4e62089bed8efbe88be81163e4d7d11c0c0/client/my-sites/checkout/checkout-thank-you/transfer-pending/index.tsx#L25-L36

It may be that we can do something like the following:
```typescript
const installedWooCommercePlugin = useSelector( ( state: AppState ) => getPluginOnSite( state, siteId, 'woocommerce' ) );
const isWooCommerceActive = installedWooCommercePlugin && installedWooCommercePlugin.active;
```
Though we would also need some kind of code to (re)trigger the fetch of plugins, especially after the transfer to Atomic has been completed. We likely want to do that via `dispatch( fetchSitePlugins( siteId ) )`, but the triggering logic may require the component to be modified.

**Comment by markbiek** [comment]
@daledupreez I went down a similar path to modifying the `TransferPending` component, with help from @nelsonec87 and @andres-blanco, and there are some issues. (Additional discussion in p1681929948724689-slack-C0Q664T29).

The main problem is that [this code](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/transfer-pending/index.tsx#L91) starting at Line 91 of `TransferPending` that checks if the transfer is complete _is never called_. 

The `CheckoutThankYou` component  is [checking the Atomic transfer status](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/index.jsx#L743) itself.

That means that `CheckoutThankYou` [discovers](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/index.jsx#L440) that the transfer is complete first and then doesn't display `TransferPending` anymore.

The reality is that the `TransferPending` component gets to about 66% on the progress bar and then it's replaced by the `ThankYouCard` meaning that there's never a time when the Atomic transfer is complete _and_ code inside `TransferPending` is running.

I do think you're correct that we need to trigger an API request to `/sites/:siteSlug/plugins` (probably, as you mentioned, with `dispatch( fetchSitePlugins( siteId ) )`).

It gets a little ugly since `CheckoutThankYou` is a class component. We discovered (thanks @andres-blanco !) that it uses the `componentDidMount` [method to call fetchAtomicTransfer](https://github.com/Automattic/wp-calypso/blob/trunk/client/my-sites/checkout/checkout-thank-you/index.jsx#L142) to check the Atomic transfer status.

We tried something similar with `fetchSitePlugins` but it didn't work. That may be because we were calling `fetchAtomicTransfer` directly rather than dispatching it.

We also wondered if it was possible to call `/sites/:siteSlug/atomic/software/woo-on-plans` instead but that seems to have the same problem this issue is describing. It seems possible for that endpoint to return `plugin/woocommerce` because WooCommerce is fully installed.

I'll be going back to playing with `fetchSitePlugins` today.
